src/main/kotlin/app/revanced/patches/all/connectivity/wifi/spoof/patch/SpoofWifiPatch.kt

```diff
@@ -0,0 +1,232 @@
+        RegisterNetworkCallback1(
+            "Landroid/net/ConnectivityManager;",
+            "registerNetworkCallback",
+            arrayOf("Landroid/net/NetworkRequest;", "Landroid/net/ConnectivityManager\$NetworkCallback;"),
+            "V",
+        ),
+        RegisterNetworkCallback2(
+            "Landroid/net/ConnectivityManager;",
+            "registerNetworkCallback",
+            arrayOf("Landroid/net/NetworkRequest;", "Landroid/app/PendingIntent;"),
+            "V",
+        ),
+        RegisterNetworkCallback3(
+            "Landroid/net/ConnectivityManager;",
+            "registerNetworkCallback",
+            arrayOf("Landroid/net/NetworkRequest;", "Landroid/net/ConnectivityManager\$NetworkCallback;", "Landroid/os/Handler;"),
+            "V",
+        ),
+        RequestNetwork1(
+            "Landroid/net/ConnectivityManager;",
+            "requestNetwork",
+            arrayOf("Landroid/net/NetworkRequest;", "Landroid/net/ConnectivityManager\$NetworkCallback;"),
+            "V",
+        ),
+        RequestNetwork2(
+            "Landroid/net/ConnectivityManager;",
+            "requestNetwork",
+            arrayOf("Landroid/net/NetworkRequest;", "Landroid/net/ConnectivityManager\$NetworkCallback;", "I"),
+            "V",
+        ),
+        RequestNetwork3(
+            "Landroid/net/ConnectivityManager;",
+            "requestNetwork",
+            arrayOf("Landroid/net/NetworkRequest;", "Landroid/net/ConnectivityManager\$NetworkCallback;", "Landroid/os/Handler;"),
+            "V",
+        ),
+        RequestNetwork4(
+            "Landroid/net/ConnectivityManager;",
+            "requestNetwork",
+            arrayOf("Landroid/net/NetworkRequest;", "Landroid/app/PendingIntent;"),
+            "V",
+        ),
+        RequestNetwork5(
+            "Landroid/net/ConnectivityManager;",
+            "requestNetwork",
+            arrayOf("Landroid/net/NetworkRequest;", "Landroid/net/ConnectivityManager\$NetworkCallback;", "Landroid/os/Handler;", "I"),
+            "V",
+        );
+
+        fun replaceInstruction(method: MutableMethod, instruction: Instruction35c, instructionIndex: Int) {
+            val registers = arrayOf(instruction.registerC, instruction.registerD, instruction.registerE, instruction.registerF, instruction.registerG)
+            val argsNum = this.methodParams.size + 1 // + 1 for instance of definedClassName
+            if (argsNum > registers.size) {
+                // should never happen, but just to be sure (also for the future) a safety check
+                throw RuntimeException("not enough registers for ${this.definedClassName}#${this.methodName}: required $argsNum registers, but only got ${registers.size}")
+            }
+            val args = registers.take(argsNum).joinToString(separator = ", ") { reg -> "v${reg}" }
+            val replacementMethodDefinition = "${this.methodName}(${this.definedClassName}${this.methodParams.joinToString(separator = "")})${this.returnType}"
```

Comment:
Using `this` adds noise to the code. Removing it turns `${this.methodName}` into `$methodName`.

Reply:
Sure, I can do that. But I’ll wait for your other responses first.

Reply:
Removing it sounds good to me.

Reply:
Done.